manifests/server/config_entry.pp

```diff
@@ -91,73 +87,6 @@
     }
   }
 
-  # We have to handle ports and the data directory in a weird and
-  # special way.  On early Debian and Ubuntu and RHEL we have to ensure
-  # we stop the service completely. On RHEL 7 we either have to create
-  # a systemd override for the port or update the sysconfig file, but this
-  # is managed for us in postgresql::server::config.
-  if $facts['os']['name'] == 'Debian' or $facts['os']['name'] == 'Ubuntu' {
```

Comment:
This is no longer relevant?

Reply:
I am pretty sure this is no longer needed. And the systemd drop-in is not managed on debian based systems. I would rely on the tests here, but I can also test a bit on a debian based system

Reply:
I really do wonder how often the data directory is changed. As I mentioned elsewhere, I'm not sure it's supported to have the data directory in postgresql.conf at all. So I think it may be ok to drop. Just a bit unsure about it.

Reply:
I did this on many debian and ubuntu systems, changing it in postgresql.conf works

Reply:
Probably because they store it in `/etc` rather than `/var` like Red Hat does (which is crazy).

Reply:
I just verified this on debian 10 with postgresql 11 (debians own package, default), changing port and data directory via postgresql.conf works fine.